File: app-miyou/src/main/java/com/miyou/app/infrastructure/inbound/web/credit/dto/UserCreditResponse.java
package com.miyou.app.infrastructure.inbound.web.credit.dto;

import com.miyou.app.domain.credit.model.UserCredit;

public record UserCreditResponse(
	String userId,
	long balance
) {
	public static UserCreditResponse from(UserCredit credit) {
		return new UserCreditResponse(credit.userId().value(), credit.balance());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/inbound/web/credit/dto/UserCreditResponse.kt
package com.miyou.app.infrastructure.inbound.web.credit.dto

import com.miyou.app.domain.credit.model.UserCredit

data class UserCreditResponse(
    val userId: String,
    val balance: Long,
) {
    companion object {
        fun from(credit: UserCredit): UserCreditResponse = UserCreditResponse(credit.userId.value, credit.balance)
    }
}
